src/txreconciliation.h

```diff
@@ -0,0 +1,585 @@
+        // the same initial set (without transactions received during the reconciliation).
+        // Allow to store new transactions separately in the original set.
+        assert(m_requestor);
+        m_capacity_snapshot = sketch_capacity;
+        m_local_set_snapshot = m_local_set;
+        m_local_set.clear();
+    }
+
+    /**
+     * TODO document
+     */
+    void PrepareForExtensionResponse(uint16_t sketch_capacity, const std::vector<uint8_t>& remote_sketch)
+    {
+        assert(m_responder);
+        m_capacity_snapshot = sketch_capacity;
+        m_remote_sketch_snapshot = remote_sketch;
+        m_local_set_snapshot = m_local_set;
+        m_local_set.clear();
+    }
+};
+
+/**
+ * Used to track reconciliations across all peers.
+ */
+class TxReconciliationTracker {
+    /**
+     * Salt used to compute short IDs during transaction reconciliation.
+     * Salt is generated randomly per-connection to prevent linking of
+     * connections belonging to the same physical node.
+     * Also, salts should be different per-connection to prevent halting
+     * of relay of particular transactions due to collisions in short IDs.
+     */
+    Mutex m_local_salts_mutex;
+    std::unordered_map<NodeId, uint64_t> m_local_salts GUARDED_BY(m_local_salts_mutex);
+
+    /**
+     * Used to keep track of ongoing reconciliations (or lack of them) per peer.
+     */
+    Mutex m_states_mutex;
+    std::unordered_map<NodeId, ReconciliationState> m_states GUARDED_BY(m_states_mutex);
+
+    /**
+     * Reconciliation should happen with peers in the same order, because the efficiency gain is the
+     * highest when reconciliation set difference is predictable. This queue is used to maintain the
+     * order of peers chosen for reconciliation.
+     */
+    Mutex m_queue_mutex;
+    std::deque<NodeId> m_queue GUARDED_BY(m_queue_mutex);
+
+    /**
+     * Reconciliations are requested periodically:
+     * every RECON_REQUEST_INTERVAL seconds we pick a peer from the queue.
+     */
+    std::chrono::microseconds m_next_recon_request{0};
+    void UpdateNextReconRequest(std::chrono::microseconds now) EXCLUSIVE_LOCKS_REQUIRED(m_queue_mutex)
+    {
+        m_next_recon_request = now + RECON_REQUEST_INTERVAL / m_queue.size();
+    }
+
+    /**
+     * Used to schedule the next initial response for any pending reconciliation request.
+     * Respond to all requests at the same time to prevent transaction possession leak.
+     */
+    std::chrono::microseconds m_next_recon_respond{0};
+    std::chrono::microseconds NextReconRespond()
+    {
+        auto current_time = GetTime<std::chrono::microseconds>();
+        if (m_next_recon_respond < current_time) {
+            m_next_recon_respond = current_time + RECON_RESPONSE_INTERVAL;
+        }
+        return m_next_recon_respond;
+    }
+
+    public:
+
+    TxReconciliationTracker() {};
+
+    /**
+     * TODO: document
+     */
```

Comment:
These interface function documents are the most important part of this refactor! I suggest you fill them to fully describe what interface this module is presenting to the client (net_processing). Once that's done, reviewers will be able to assess whether you're proposing a sensible interface.